### PR TITLE
Created add service account roles assign screen

### PR DESCRIPTION
**Pull request by edewit**
## Motivation

Service account roles assign screen as designed [marvel](https://marvelapp.com/prototype/f058374/screen/71931944)

## Brief Description

Couldn't put the filter dropdown next to the search filter as the select needs to be a direct child of the modal.
Maybe somebody with some more css skills can have a look

## Verification Steps

Create a service account client and go to the service account tab and press assign

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented
- [x] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

File: src/clients/service-account/AddServiceAccountModal.tsx [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  return (
+    <Modal
+      variant={ModalVariant.large}
+      title={t("assignRolesTo", { client: clientId })}
+      isOpen={true}
+      onClose={onClose}
+      actions={[
+        <Button
+          data-testid="assign"
+          key="confirm"
```
Comment (jenny-s51) [inline src/clients/service-account/AddServiceAccountModal.tsx:180]:
```suggestion
          isDisabled={selectedRows?.length === 0}
```

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
LGTM @edewit - role assignment works as expected! Just one small suggested update to disable the "Assign" button if no roles are selected ... currently an error gets thrown if you click "Assign" without selecting anything.

**Comment by edewit** [comment]
thanks @jenny-s51

**Review by ssilvert — APPROVED** [review]
I'm going to go ahead and merge this before the holiday weekend.  I'll open some issues for things I think should be addressed.

**Pull request merged**